### ISSUE TITLE
Why Use AUTOINCREMENT Keyword?

### ISSUE THREAD

**Issue by scout208** [MicrosoftDocs/windows-uwp#657]
In the command to create the table, you write 

    String tableCommand = "CREATE TABLE IF NOT " +
                "EXISTS MyTable (Primary_Key INTEGER PRIMARY KEY AUTOINCREMENT, " +
                "Text_Entry NVARCHAR(2048) NULL)";

I don't understand why you use the AUTOINCREMENT keyword here. According to the [SQLite documentation](https://www.sqlite.org/autoinc.html),

>1. The AUTOINCREMENT keyword imposes extra CPU, memory, disk space, and disk I/O overhead and should be avoided if not strictly needed. It is usually not needed.
>2. In SQLite, a column with type INTEGER PRIMARY KEY is an alias for the ROWID (except in WITHOUT ROWID tables) which is always a 64-bit signed integer.
>3. On an INSERT, if the ROWID or INTEGER PRIMARY KEY column is not explicitly given a value, then it will be filled automatically with an unused integer, usually one more than the largest ROWID currently in use. This is true regardless of whether or not the AUTOINCREMENT keyword is used.
>4. If the AUTOINCREMENT keyword appears after INTEGER PRIMARY KEY, that changes the automatic ROWID assignment algorithm to prevent the reuse of ROWIDs over the lifetime of the database. In other words, the purpose of AUTOINCREMENT is to prevent the reuse of ROWIDs from previously deleted rows.

Based on that, I don't see why it is necessary to use AUTOINCREMENT here.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: deb3f64f-cb34-b6e2-a9a0-6ed8eb741cb3
* Version Independent ID: c8f31cef-b782-76ff-df62-9f26c54e8897
* Content: [Use a SQLite database in a UWP app - UWP app developer](https://docs.microsoft.com/en-us/windows/uwp/data-access/sqlite-databases#some-benefits-of-using-sqlite-for-local-storage)
* Content Source: [windows-apps-src/data-access/sqlite-databases.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/data-access/sqlite-databases.md)
* Product: **windows**
* GitHub Login: @normesta
* Microsoft Alias: **normesta**

**Comment by normesta** [comment]
Thank  you for the input. I've looked into this a bit. This post confirms your finding (http://www.sqlitetutorial.net/sqlite-autoincrement/) stating that the only valid use of this attribute is to "o prevent SQLite to reuse value that has not been used or from the previously deleted row.". I tried removing it from the query and all works just as fine. I've removed the attribute from the example and it should appear in the live version of this doc over the next day or so. Thank you!
